tezedge_state/src/proposer/proposal_persister.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) SimpleStaking, Viable Systems and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+use std::fs::File;
+use std::io::prelude::*;
+use std::io::BufWriter;
+use std::sync::mpsc;
+
+use crate::proposals::*;
+
+pub struct ProposalPersisterHandle {
+    sender: Option<mpsc::SyncSender<RecordedProposal>>,
+    join_handle: Option<std::thread::JoinHandle<()>>,
+}
+
+impl Drop for ProposalPersisterHandle {
+    fn drop(&mut self) {
+        drop(self.sender.take());
+        if let Some(jh) = self.join_handle.take() {
+            jh.join().unwrap();
+        }
+    }
+}
+
+impl Clone for ProposalPersisterHandle {
+    fn clone(&self) -> Self {
+        // TODO
+        unimplemented!("this is temporary, this type shouldn't be clonable")
+    }
+}
+
+impl ProposalPersisterHandle {
+    pub fn persist<P>(&mut self, proposal: P)
+    where
+        P: Into<RecordedProposal>,
+    {
+        // TODO: better error handling.
+        self.sender
+            .as_mut()
+            .unwrap()
+            .send(proposal.into())
+            .expect("recorded proposal persister disconnected!");
+    }
+}
+
+pub struct ProposalPersister {
+    channel: mpsc::Receiver<RecordedProposal>,
+}
+
+impl ProposalPersister {
+    pub fn start() -> ProposalPersisterHandle {
+        let (tx, rx) = mpsc::sync_channel(1024);
+
+        let state = Self { channel: rx };
+
+        let join_handle = std::thread::spawn(move || run(state));
```

Comment:
@binier 
please spawn this with thread name like we do everywhere for std threads,
also this makes you propage error, `-> Result<ProposalPersisterHandle, Error>` :)

e.g.:
```
let proposer_thread_handle = std::thread::Builder::new()
                .name("tezedge-proposer".to_owned())
                .spawn(move || {
                    proposer.extend_potential_peers(Instant::now(), initial_potential_peers);
                    run(
                        proposer,
                        proposer_rx,
                        network_event_notifier,
                        shell_command_notifier,
                    );
                })
                .map_err(TezedgeStateManagerSpawnError::IoError)?;
```